middlewares/prometheus_test.go

```diff
@@ -42,6 +44,81 @@ func TestPrometheus(t *testing.T) {
 		t.Errorf("body does not contain request total entry '%s'", reqsName)
 	}
 	if !strings.Contains(body, latencyName) {
-		t.Errorf("body does not contain request duration entry '%s'", reqsName)
+		t.Errorf("body does not contain request duration entry '%s'", latencyName)
+	}
+
+	// Register the same metrics again
+	metricsMiddlewareBackend = NewMetricsWrapper(NewPrometheus("test", &types.Prometheus{}))
+	n = negroni.New()
+	n.Use(metricsMiddlewareBackend)
+	n.UseHandler(r)
+
+	n.ServeHTTP(recorder, req2)
+
+	metricsFamily, err := prometheus.DefaultGatherer.Gather()
+	if err != nil {
+		t.Error(err)
+	}
+
+	var latencyMetricFamily, reqsMetricFamily *dto.MetricFamily
+
+	for _, metricFamily := range metricsFamily {
+		if metricFamily.GetName() == reqsName {
+			reqsMetricFamily = metricFamily
+		}
+		if metricFamily.GetName() == latencyName {
+			latencyMetricFamily = metricFamily
+		}
+	}
+
+	if reqsMetricFamily == nil {
+		t.Errorf("gathered metrics does not contain request total entry '%s'", reqsName)
+	}
+
+	if latencyMetricFamily == nil {
+		t.Errorf("gathered metrics does not contain request duration entry '%s'", latencyName)
+	}
+
+	expectedLabelsForMetricFamily := []struct {
+		name         string
+		metricFamily *dto.MetricFamily
+		labels       map[string]string
+	}{
+		{
+			reqsName,
+			reqsMetricFamily,
+			map[string]string{
+				"code":    "200",
+				"method":  "GET",
+				"service": "test",
+			},
+		},
+		{
+			latencyName,
+			latencyMetricFamily,
+			map[string]string{
+				"service": "test",
+			},
+		},
+	}
```

Comment:
Right after the definition of the test cases, I think we should do a check between the number of metric families we found and the number of test cases we have. This way, we can catch the situation where have added a new metric without updating the test.